src/content/layout/builtin_layout.cc

```diff
@@ -362,93 +380,112 @@ void BuiltinLayout::addAudio(const std::shared_ptr<CdsObject>& obj, const std::s
     }
 
     auto artistContainer = std::make_shared<CdsContainer>(artist, UPNP_CLASS_MUSIC_ARTIST);
-    std::vector<std::shared_ptr<CdsObject>> arc;
-    arc.push_back(container["Audio"]);
-    arc.push_back(container["Artists"]);
-    arc.push_back(artistContainer);
-    arc.push_back(container["All Songs"]);
-    id = content->addContainerTree(arc, obj);
-    add(obj, id);
-
-    std::string temp;
+    if (blOption->get("Audio/allSongs")->getEnabled()) {
+        std::vector<std::shared_ptr<CdsObject>> arc;
+        arc.push_back(container["Audio"]);
+        arc.push_back(container["Artists"]);
+        arc.push_back(artistContainer);
+        arc.push_back(container["All Songs"]);
+        id = content->addContainerTree(arc, obj);
+        add(obj, id);
+    }
+
+    std::string prefixTitle;
     if (!artistFull.empty())
-        temp = artistFull;
+        prefixTitle = artistFull;
 
     if (!albumFull.empty())
-        temp = fmt::format("{} - {} - ", temp, albumFull);
+        prefixTitle = fmt::format("{} - {} - ", prefixTitle, albumFull);
     else
-        temp = fmt::format("{} - ", temp);
+        prefixTitle = fmt::format("{} - ", prefixTitle);
 
     auto albumContainer = std::make_shared<CdsContainer>(album, UPNP_CLASS_MUSIC_ALBUM);
     albumContainer->setMetaData(obj->getMetaData());
     albumContainer->setResources(obj->getResources());
     albumContainer->setRefID(obj->getID());
     artistContainer->setSearchable(true);
 
-    std::vector<std::shared_ptr<CdsObject>> alc;
-    alc.push_back(container["Audio"]);
-    alc.push_back(container["Artists"]);
-    alc.push_back(artistContainer);
-    alc.push_back(albumContainer);
-    id = content->addContainerTree(alc, obj);
-    add(obj, id);
+    if (blOption->get("Audio/allArtists")->getEnabled()) {
+        std::vector<std::shared_ptr<CdsObject>> alc;
+        alc.push_back(container["Audio"]);
+        alc.push_back(container["Artists"]);
+        alc.push_back(artistContainer);
+        alc.push_back(albumContainer);
+        id = content->addContainerTree(alc, obj);
+        add(obj, id);
+    }
 
     albumContainer->setSearchable(true);
```

Comment:
This should be in the if case as well

Reply:
I did left it intentionately, because it could (it is not) be used in any of the further conditions... and then we would build a "cross-dependency" between containers